mega-service-agentQnA/validate_agents.sh

```
@@ -0,0 +1,68 @@
+if [ $? -eq 0 ]; then
+  echo "✅ Supervisor Agent test passed"
+else
+  echo "❌ Supervisor Agent test failed"
+fi
+
+echo -e "\n================================================================="
+echo "                      VALIDATION COMPLETE                           "
+echo "================================================================="
+
+# Check if all tests passed
+if [ $? -eq 0 ]; then
```

Shellcheck (0.10.0)

[warning] This $? refers to echo/printf, not a previous command. Assign to variable to avoid it being overwritten.

(SC2320)